test/integration/context-based-restrictions.v1.test.js

```diff
@@ -0,0 +1,934 @@
+    expect(res.status).toBe(200);
+    expect(res.result).toBeDefined();
+  });
+
+  test('getZone() - 400 - missing required parameter: zoneId', async () => {
+    // The zone could not be found due to missing zone ID.
+
+    const params = {
+      transactionId: uuidv4(),
+    };
+
+    await expect(contextBasedRestrictionsService.getZone(params)).rejects.toMatchObject({
+      'message': 'Missing required parameters: zoneId',
+    });
+  });
+
+  test('getZone() - 404 - unexisting zone ID', async () => {
+    // The zone could not be found due to unexisting zone ID.
+
+    const params = {
+      zoneId: unexistingZoneId,
+      transactionId: uuidv4(),
+    };
+
+    await expect(contextBasedRestrictionsService.getZone(params)).rejects.toMatchObject({
+      status: 404,
+    });
+  });
+
+  test('replaceZone() - 200', async () => {
+    // Request models needed by this operation.
+
+    // AddressIPAddress
+    const addressModel = {
+      type: 'ipAddress',
+      value: '169.24.57.235',
+    };
+
+    const params = {
+      zoneId,
+      ifMatch: zoneEtag,
+      name: 'an example of updated zone',
+      accountId,
+      addresses: [addressModel],
+      description: 'this is an example of updated zone',
+      transactionId: uuidv4(),
+    };
+
+    const res = await contextBasedRestrictionsService.replaceZone(params);
+    expect(res).toBeDefined();
+    expect(res.status).toBe(200);
+    expect(res.result).toBeDefined();
+  });
+
+  test('replaceZone() - 400 - invalid type: Address', async () => {
+    // The zone could not be updated due to invalid type 'Address'.
+    // Request models needed by this operation.
+
+    // AddressIPAddress
+    const addressModel = {
+      type: 'Address',
+      value: '169.24.57.235',
+    };
+
+    const params = {
+      zoneId,
+      ifMatch: zoneEtag,
+      name: 'an example of an invalid zone',
+      accountId,
+      addresses: [addressModel],
+      description: 'this is an example of an invalid zone',
+      transactionId: uuidv4(),
+    };
+
+    await expect(contextBasedRestrictionsService.replaceZone(params)).rejects.toMatchObject({
+      status: 400,
+    });
+  });
+
+  test('replaceZone() - 400 - missing required parameter: ifMatch', async () => {
+    // The zone could not be updated due to missing required parameter: ifMatch'.
+    // Request models needed by this operation.
+
+    // AddressIPAddress
+    const addressModel = {
+      type: 'Address',
+      value: '169.24.57.235',
+    };
+
+    const params = {
+      zoneId,
+      name: 'an example of an invalid zone',
+      accountId,
+      addresses: [addressModel],
+      description: 'this is an example of an invalid zone',
+      transactionId: uuidv4(),
+    };
+
+    await expect(contextBasedRestrictionsService.replaceZone(params)).rejects.toMatchObject({
+      'message': 'Missing required parameters: ifMatch',
+    });
+  });
+
+  test('replaceZone() - 400 - zone not found', async () => {
```

Comment:
I assume this test is about `404` and not `400`.

Reply:
good catch, changed to 404